weather_app.py

```diff
@@ -0,0 +1,50 @@
+import requests
+
+# Data Source Class
+class HourlyTemperature:
+    def __init__(self, temperatures: list):
+        self.temperatures = temperatures
```

Comment:
you should also add a field to the dates/timestamps so that each temperature value has an associated point in time for better visualization